index_configuration_tool/tests/test_search_endpoint.py

```diff
@@ -0,0 +1,66 @@
+import copy
+import unittest
+
+import responses
+from responses import matchers
+
+import search_endpoint
+
+# Constants
+TEST_ENDPOINT = "http://test/"
+INDEX1_NAME = "index1"
+INDEX2_NAME = "index2"
+SETTINGS_KEY = "settings"
+INTERNAL_INDEX_KEY = "index"
+TEST_DATA = {
+    INDEX1_NAME: {
+        SETTINGS_KEY: {
+            INTERNAL_INDEX_KEY: {
+                "uuid": "test",
+                "version": 1
+            },
+            "bool_key": True,
+            "str_array": ["abc", "x y z"],
+        },
+        "mappings": {
+            "str_key": "string value",
+            "num_key": -1
+        }
+    },
+    INDEX2_NAME: {
+        SETTINGS_KEY: {},
+        "mappings": {}
+    }
+}
+
+
+class TestSearchEndpoint(unittest.TestCase):
+    def setUp(self) -> None:
+        self.test_data_without_internal = copy.deepcopy(TEST_DATA)
+        # Remove internal data
+        for i in self.test_data_without_internal.keys():
+            self.test_data_without_internal[i][SETTINGS_KEY].pop(INTERNAL_INDEX_KEY, None)
+
+    @responses.activate
```

Comment:
This is cool! I haven't seen the `responses` library before, but it looks super helpful!